src/Descriptor.cpp

```diff
@@ -0,0 +1,113 @@
+#include "Descriptor.hpp"
+
+#include "comm.hpp"
+#include "string_utils.hpp"
+
+// Up to 5 characters
+const char *short_name_of(DescriptorState state) {
+    switch (state) {
+    case DescriptorState::Playing: return "Play";
+    case DescriptorState::GetName: return "Name";
+    case DescriptorState::GetOldPassword: return "OldPw";
+    case DescriptorState::ConfirmNewName: return "CnfNm";
+    case DescriptorState::GetNewPassword: return "NewPw";
+    case DescriptorState::ConfirmNewPassword: return "CnfNp";
+    case DescriptorState::GetNewRace: return "GetNR";
+    case DescriptorState::GetNewSex: return "GetNS";
+    case DescriptorState::GetNewClass: return "GetNA";
+    case DescriptorState::GetAlignment: return "GetAl";
+    case DescriptorState::DefaultChoice: return "DefCh";
+    case DescriptorState::GenGroups: return "GenGr";
+    case DescriptorState::ReadIMotd: return "Imotd";
+    case DescriptorState::ReadMotd: return "Motd";
+    case DescriptorState::BreakConnect: return "BrkCn";
+    case DescriptorState::GetAnsi: return "Ansi";
+    case DescriptorState::CircumventPassword: return "CPass";
+    case DescriptorState::Disconnecting: return "Disc";
+    case DescriptorState::DisconnectingNp: return "DiscN";
+    }
+    return "<UNK>";
+}
+
+const char *name_of(DescriptorState state) {
+    switch (state) {
+    case DescriptorState::Playing: return "Playing";
```

Comment:
Personally, I feel this is one of those rare moments when a macros makes the code better:

`#define CS(x) case DescriptorState::x: return #x;`

That way, you can't accidentally end up with something inconsistent (by copy/paste/imperfect-hacking).

That said, this entire function isn't used anywhere, so I'd suggest killing it. If the short name is all we need, then let's go with just that.

Reply:
Yeah, agreed. I sometimes use macros; but IDE support makes this pretty easy (refactors rename the strings too). Agreed re:killing it off, it's habit and a hedge for a future world in which I want to make sure (for debugging) we can stream out any enums etc. But...will remove.